app/src/sharedTest/java/org/oppia/android/app/player/state/StateFragmentTest.kt

```diff
@@ -206,17 +207,184 @@ class StateFragmentTest {
+      clickSubmitAnswerButton()
+
+      rotateToLandscape()
+
+      scrollToViewType(FEEDBACK)
+      onView(withId(R.id.feedback_text_view)).check(matches(isDisplayed()))
+      onView(withId(R.id.fraction_input_interaction_view)).check(matches(isDisplayed()))
+    }
+  }
+
   //  6. Backward/forward navigation along with configuration changes to verify that you stay on the
   //     navigated state.
+  @Test
+  fun testStateFragment_loadExp_moveForward_changeConfiguration_stayOnTheNavigatedState() {
+    launchForExploration(TEST_EXPLORATION_ID_2).use {
+      startPlayingExploration()
+      clickContinueInteractionButton()
+
+      typeFractionText("1/2")
+      clickSubmitAnswerButton()
+      clickContinueNavigationButton()
+
+      rotateToLandscape()
+      scrollToViewType(CONTENT)
+      val htmlResult =
+        "Which bird can sustain flight for long periods of time?"
+      onView(atPositionOnView(R.id.state_recycler_view, 0, R.id.content_text_view)).check(
+        matches(
+          withText(htmlResult)
+        )
+      )
+    }
+  }
+
+  @Test
+  fun testStateFragment_loadExp_moveForward_moveBackward_changeConfiguration_stayOnSameState() {
+    launchForExploration(TEST_EXPLORATION_ID_2).use {
+      startPlayingExploration()
+      clickContinueInteractionButton()
+
+      typeFractionText("1/2")
+      clickSubmitAnswerButton()
+      clickContinueNavigationButton()
+
+      clickPreviousNavigationButton()
+
+      rotateToLandscape()
+      scrollToViewType(CONTENT)
+      val htmlResult =
+        "What fraction represents half of something?"
+      onView(atPositionOnView(R.id.state_recycler_view, 0, R.id.content_text_view)).check(
+        matches(
+          withText(htmlResult)
+        )
+      )
+    }
+  }
+
   //  7. Verifying that old answers were present when navigation backward/forward.
+  @Test
+  fun testStateFragment_loadExp_secondState_submitAnswer_moveForward_moveBackward_answerVisible() {
+    launchForExploration(TEST_EXPLORATION_ID_2).use {
+      startPlayingExploration()
+      clickContinueInteractionButton()
+
+      typeFractionText("1/2")
+      clickSubmitAnswerButton()
+      clickContinueNavigationButton()
+
+      clickPreviousNavigationButton()
+
+      scrollToViewType(SUBMITTED_ANSWER)
+      onView(withId(R.id.submitted_answer_text_view)).check(
+        matches(
+          withText("1/2")
+        )
+      )
+    }
+  }
+
   //  8. Testing providing the wrong answer and showing hints.
+  @Test
+  @Ignore("Failing")
+  fun testStateFragment_loadExp_fractionState_submitWrongAnswerTwice_hintsVisible() {
+    launchForExploration(FRACTIONS_EXPLORATION_ID_1).use {
+      startPlayingExploration()
+      selectMultipleChoiceOption(optionPosition = 2)
+      selectMultipleChoiceOption(optionPosition = 2)
+
+      onView(withId(R.id.hints_and_solution_fragment_container)).check(matches(isDisplayed()))
+    }
+  }
+
   //  9. Testing all possible invalid/error input cases for each interaction.
```

Comment:
Are we covering all the interactions here?

Reply:
For point 8, currently no, as mentioned in https://github.com/oppia/oppia-android/pull/2798#discussion_r594045746

Reply:
I mean in the point `// 9. Testing all possible invalid/error input cases for each interaction.` how many types of input are we testing. 

Like in the `testStateFragment_loadExp_sixthState_submitInvalidAnswer_disablesSubmitAndShowsError`, here we testing `Ratio` type inputs. 
Can we write test for other input types like fractions etc.